# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../iproute2/xtables.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./tc/m_xt.c.vanella	2011-08-12 11:12:44.316986453 -0400
+++ ./tc/m_xt.c	2011-08-12 11:17:48.533484052 -0400
@@ -162,7 +162,11 @@
 					return -1;
 				}
 				tcipt_globals.opts =
-				    xtables_merge_options(tcipt_globals.opts,
+				    xtables_merge_options(
+#if (XTABLES_VERSION_CODE >= 6)
+                                                          tcipt_globals.orig_opts,
+#endif
+                                                          tcipt_globals.opts,
 				                          m->extra_opts,
 				                          &m->option_offset);
 			} else {
@@ -307,7 +311,11 @@
 			}
 
 			tcipt_globals.opts =
-			    xtables_merge_options(tcipt_globals.opts,
+			    xtables_merge_options(
+#if (XTABLES_VERSION_CODE >= 6)
+                                                  tcipt_globals.orig_opts,
+#endif
+                                                  tcipt_globals.opts,
 			                          m->extra_opts,
 			                          &m->option_offset);
 		} else {

